pages/api/sitemap.js

```diff
@@ -0,0 +1,28 @@
+const globby = require('globby');
+
+export default async (_, res) => {
+    res.setHeader('Content-Type', 'text/xml');
+
+    const pages = await globby(['pages/**/*{.js,.mdx}', '!pages/_*.js', '!pages/api']);
+    const sitemap = `<?xml version="1.0" encoding="UTF-8"?>
+        <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+            ${pages
+                .map((page) => {
+                    const path = page
+                        .replace('pages', '')
+                        .replace('.js', '')
+                        .replace('.mdx', '');
+                    const route = path === '/index' ? '/' : path;
+
+                    return `
+                        <url>
+                            <loc>${`https://manovotny.com${route}`}</loc>
+                        </url>
+                    `;
+                })
+                .join('')}
+        </urlset>
+    `;
+
+    res.status(200).send(sitemap);
+};
```

Comment:
@leerob wouldn't this code run every time the endpoint is hit? Seems like it should still be generated once and just be a static asset.

Reply:
It would. I think there should probably be caching on it. Something like...

```json
{
    "src": "/api/sitemap",
    "dest": "/api/sitemap",
    "headers": {"cache-control": "public,max-age=120,stale-while-revalidate=60"}
}
```

Maybe it updates every day? Side note: this script didn't work deployed. After compiling, there's no `/pages` folder 🤦‍♂ So I'm kind of up in the air on this approach. It could modify this to look for the `.html` files with globby. Or I might revert back to an after build script. Considering there's also a list of articles in the data folder, you could also use that to dynamically build the sitemap + the static routes you have.

Reply:
Checkout this beaut! 😍 

https://github.com/manovotny/manovotny.com/pull/94/commits/42abd010f9ce78992e248f91afb355401451c93b